fe/fe-core/src/main/java/com/starrocks/sql/analyzer/InsertAnalyzer.java

```diff
@@ -286,7 +263,102 @@ public static void analyze(InsertStmt insertStmt, ConnectContext session) {
         insertStmt.setTargetTable(table);
         insertStmt.setTargetPartitionIds(targetPartitionIds);
         insertStmt.setTargetColumns(targetColumns);
-        session.getDumpInfo().addTable(database.getFullName(), table);
+        session.getDumpInfo().addTable(insertStmt.getTableName().getDb(), table);
+    }
+
+    private static Table getOrMakeTargetTable(InsertStmt insertStmt, ConnectContext session) {
+        if (insertStmt.useTableFunctionAsTargetTable()) {
+            return makeTableFunctionTable(insertStmt);
+        }
+
+        MetaUtils.normalizationTableName(session, insertStmt.getTableName());
+        String catalogName = insertStmt.getTableName().getCatalog();
+        String dbName = insertStmt.getTableName().getDb();
+        String tableName = insertStmt.getTableName().getTbl();
+
+        try {
+            MetaUtils.checkCatalogExistAndReport(catalogName);
+        } catch (AnalysisException e) {
+            ErrorReport.reportSemanticException(ErrorCode.ERR_BAD_CATALOG_ERROR, catalogName);
+        }
+
+        Table table = MetaUtils.getTable(catalogName, dbName, tableName);
+
+        if (table instanceof ExternalOlapTable) {
+            Database database = MetaUtils.getDatabase(catalogName, dbName);
+            table = getOLAPExternalTableMeta(database, (ExternalOlapTable) table);
+        }
+        return table;
+    }
+
+    private static Table makeTableFunctionTable(InsertStmt insertStmt) {
+        // fetch schema from query
+        QueryRelation query = insertStmt.getQueryStatement().getQueryRelation();
+        List<Field> allFields = query.getRelationFields().getAllFields();
+        List<Column> columns = allFields.stream().filter(Field::isVisible).map(field -> new Column(field.getName(),
+                field.getType(), field.isNullable())).collect(Collectors.toList());
+
+        // parse table function properties
+        Map<String, String> props = insertStmt.getTableFunctionProperties();
+        boolean writeSingleFile = Boolean.parseBoolean(props.get("single"));
+        String path = props.get("path");
+        String format = props.get("format");
+        String partitionBy = props.get("partition_by");
+
+        // validate properties
+        if (path == null) {
+            throw new SemanticException(
+                    "path is mandatory in table function. \"path\" = \"hdfs://path/to/your/location/prefix\"");
+        }
+
+        if (format == null) {
+            throw new SemanticException("format is mandatory in table function. " +
+                    "Use \"path\" = \"parquet\" as only parquet format is supported now");
+        }
+
+        if (!format.equalsIgnoreCase("parquet")) {
```

Comment:
credential params and file size params are missing. you need to implement based on the PRD.